core/blockchain.go

```diff
@@ -298,13 +300,14 @@ type BlockChain struct {
 // NewBlockChain returns a fully initialised block chain using information
 // available in the database. It initialises the default Ethereum Validator and
 // Processor.
-func NewBlockChain(
-	db ethdb.Database, cacheConfig *CacheConfig, chainConfig *params.ChainConfig, engine consensus.Engine,
-	vmConfig vm.Config, lastAcceptedHash common.Hash,
-) (*BlockChain, error) {
+func NewBlockChain(db ethdb.Database, cacheConfig *CacheConfig, chainConfig *params.ChainConfig, engine consensus.Engine, vmConfig vm.Config, lastAcceptedHash common.Hash, backend precompile.Backend) (*BlockChain, error) {
 	if cacheConfig == nil {
 		return nil, errCacheConfigNotSpecified
 	}
+	// TODO: This is added to handle the nil backends passed in from test, when warp is initialized we should remove
+	if backend == nil {
+		backend = precompile.NewNoopBackend()
+	}
```

Comment:
Can we remove this now?

Reply:
We should define an invariant for when this needs to be populated vs. when it can be nil.

For existing tests that do not touch warp, it should be safe to leave this as nil.

Additionally, if we are re-processing blocks that have already been accepted (in which case we will not call Accept on them and the precompile's `onAccept` functionality), it should also be safe for this to be left nil.